envoy: Pass --experimental_strict_action_env to bazel test as well.

jrajahalme opened the pull request:
Otherwise `make tests` invalidates the bazel cache, also for a subsequest `make`.

Reported-by: Ray Bejjani <ray@covalent.io>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

raybejjani reviewed (approved):
Should we just include BAZEL_BUILD_OPTS on the test line directly? this way if we change any build options the tests reflect that.
Separately, do we expect to never run tests with PKG_BUILD=1 ?

jrajahalme committed:
envoy: Pass BAZEL_BUILD_OPTS to bazel test as well.

Otherwise `make tests` invalidates the bazel cache, also for a subsequest `make`.

Reported-by: Ray Bejjani <ray@covalent.io>
Signed-off-by: Jarno Rajahalme <jarno@covalent.io>
